Source/Core/DolphinQt/Achievements/AchievementProgressWidget.cpp

```diff
@@ -49,6 +51,45 @@ AchievementProgressWidget::AchievementProgressWidget(QWidget* parent) : QWidget(
 QGroupBox*
 AchievementProgressWidget::CreateAchievementBox(const rc_api_achievement_definition_t* achievement)
 {
+  QLabel* a_badge = new QLabel();
+  const auto unlock_status = AchievementManager::GetInstance()->GetUnlockStatus(achievement->id);
```

Comment:
This crashes if there's no entry in the unlock map for the given achievement, which currently happens very easily on game shutdown because you're clearing the unlock map but orphaning the game data. I'd suggest making this more tolerant, and I'd also suggest actually clearing the game data on shutdown so algorithms can't work on stale data.

Reply:
Yeah, you landed on one of the bugs I'm fixing in that other PR. Hopefully this will be fixed properly there.

Reply:
See https://github.com/dolphin-emu/dolphin/pull/12204